### PR TITLE
[ProxyManager] fix closure binding

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #33037
| License       | MIT
| Doc PR        | -

Follows #32992

### CODE REVIEW

File: src/Symfony/Bridge/ProxyManager/LazyProxy/PhpDumper/ProxyDumper.php [hunk 1]
```diff
@@ -113,7 +113,7 @@ public function getProxyCode(Definition $definition)
         }
 
         if (version_compare(self::getProxyManagerVersion(), '2.5', '<')) {
-            $code = str_replace(' \Closure::bind(function ', ' \Closure::bind(static function ', $code);
+            $code = preg_replace(' \Closure::bind(function ((?:& )?\(\$instance(?:, \$value)?\))', ' \Closure::bind(static function \1', $code);
```
Comment (Devristo) [inline src/Symfony/Bridge/ProxyManager/LazyProxy/PhpDumper/ProxyDumper.php]: This gives me the following error on:

```
In ProxyDumper.php line 116:

  [ErrorException]
  Warning: preg_replace(): Delimiter must not be alphanumeric or backslash
```

Comment (nicolas-grekas) [reply src/Symfony/Bridge/ProxyManager/LazyProxy/PhpDumper/ProxyDumper.php]: oops, fixed

Comment (Devristo) [reply src/Symfony/Bridge/ProxyManager/LazyProxy/PhpDumper/ProxyDumper.php]: I think there is still something wrong:

```
In ProxyDumper.php line 116:

  [ErrorException]
  Warning: preg_replace(): Compilation failed: missing closing parenthesis at offset 62
```

The parenthesis are not properly escaped.

File: src/Symfony/Bridge/ProxyManager/LazyProxy/PhpDumper/ProxyDumper.php [hunk 2]
```diff
@@ -113,7 +113,7 @@ public function getProxyCode(Definition $definition)
         }
 
         if (version_compare(self::getProxyManagerVersion(), '2.5', '<')) {
-            $code = str_replace(' \Closure::bind(function ', ' \Closure::bind(static function ', $code);
+            $code = preg_replace('/ \Closure::bind(function ((?:& )?\(\$instance(?:, \$value)?\))/', ' \Closure::bind(static function \1', $code);
```
Comment (Devristo) [inline src/Symfony/Bridge/ProxyManager/LazyProxy/PhpDumper/ProxyDumper.php]: I think there is still something wrong:

```
In ProxyDumper.php line 116:

  [ErrorException]
  Warning: preg_replace(): Compilation failed: missing closing parenthesis at offset 62
```

The parenthesis are not properly escaped.

Comment (nicolas-grekas) [reply src/Symfony/Bridge/ProxyManager/LazyProxy/PhpDumper/ProxyDumper.php]: better now?

Comment (Devristo) [reply src/Symfony/Bridge/ProxyManager/LazyProxy/PhpDumper/ProxyDumper.php]: Yes! Cant confirm whether the related bug is fixed as it seems I am hitting a different bug than the reporter.

**Pull request merged**